### PR TITLE
Issue #2: Deck struct

### LINKED ISSUE #2: Vytvořit Deck
Prerekvizita #1 

Interně implementován pomocí lízacího a odhazovacího balíčku

- [x] má funkci `new()`, kterou je vytvořen balíček s kartami podle

text
(https://upload.wikimedia.org/wikipedia/commons/thumb/9/95/UNO_cards_deck.svg/1024px-UNO_cards_deck.svg.png)
- [x] má funkci `draw() -> Card` 
*Karta je sebrána z vrchu lízacího balíčku. Pokud je lízací balíček prázdný, vezmou se všechny karty z odhazovacího balíčku, zamíchají se, a stanou se lízacím balíčkem.*

- [x] má funkci `play(Card)`
*Karta je uložena na vrch odhazovacího balíčku.*

### PR DESCRIPTION
**Pull request by AurumTheEnd**
Interně implementován pomocí lízacího a odhazovacího balíčku

- [x] má funkci `new()`, kterou je vytvořen lízací balíček s kartami (fcd0c07d58265e72c6184868312633f3aba0c738)
- [x] má funkci `draw() -> Card` (b32912de0e7df8616763c4d1343a099ae7ef18f2)
*Karta je sebrána z vrchu lízacího balíčku. Pokud je lízací balíček prázdný, vezmou se všechny karty z odhazovacího balíčku, zamíchají se, a stanou se lízacím balíčkem.*

- [x] má funkci `play(Card)` (6e2ad68904711ea873146dd90c26b9a34cc54630)
*Karta je uložena na vrch odhazovacího balíčku.*
 
---
Has #3 as basis and prerequisite, will rebase after that is merged to keep only the relevant commits.

### CODE REVIEW

File: backend/src/cards/deck.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                        if parity == 0 {
+                            CardSymbol::Wild
+                        } else {
+                            CardSymbol::Draw4
+                        },
+                    )
+                    .unwrap(),
+                )
+            }
+        }
```
Comment (holubond) [inline backend/src/cards/deck.rs]: Za mě jsou asi lepší hloupá, ale dobře čitelná řešení, význam `parity` mi chvíli trval pochopit :) Ale tady je to ještě v pohodě, takže to tak klidně můžeš nechat

File: backend/src/cards/deck.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    fn shuffle_draw_pile(&mut self) {
+        self.draw_pile.shuffle(&mut rand::thread_rng());
+    }
+
+    pub fn draw(&mut self) -> Card {
+        match self.draw_pile.pop() {
+            None => {
+                if self.discard_pile.is_empty() {
+                    // what should happen when the players hold all 108 cards, leaving both piles empty?
+                    panic!("No cards left in either the draw or discard pile!")
```
Comment (holubond) [inline backend/src/cards/deck.rs]: TODO dle výsledu debaty na DC

File: backend/src/cards/deck.rs (hunk 1 above)
Comment (AurumTheEnd) [reply backend/src/cards/deck.rs]: Yeah, I didn't know what else to call it... but you get 2 of most cards, exception being 0 and wild/draw4, so it made sense to have the 0..2 for-loop.

Comment (holubond) [reply backend/src/cards/deck.rs]: The clean solution (in my opinion, as these things are quite subjective) would be to break it down into several functions (e.g. `insert_black_cards()` and `insert_colour_cards()`), maybe use `.map()` for colours, get rid of parity.

Generally you want to write code that can be easily modified in order to satisfy a sudden change in the task without having to rewrite many lines. E.g. we could want to have a few more wildcards in the deck.

The logic of creating a deck of cards is relatively simple and cleaning up is not really necessary here, but having a similarly complex logic for a non-trivial feature could be devastating if a requirement changes - and in practice, they change quite a lot :)

**Review by holubond — APPROVED** [review]

File: backend/src/cards/deck.rs (hunk 1 above)
Comment (AurumTheEnd) [reply backend/src/cards/deck.rs]: Amended into f0a1c0e, force-pushed.

**Review by holubond — APPROVED** [review]

**Pull request merged**